docs/the-new-architecture/pillars-codegen.md

```diff
@@ -230,3 +230,59 @@ The **Codegen** for a Fabric Native Component contains a `MyFabricComponentManag
 
 The **Codegen** also creates a `ComponentDescriptor.h`, which is required to get a proper handle on the Fabric Native Component.
+
+## The Codegen CLI
+
+**npx react-native codegen** [**--path** *path*] [**--platform** *string*] [**--outputPath** *path*]
+
+#### DESCRIPTION
+
+This command runs `react-native-codegen` for your project.
+The codegen must be configured as described in [Configure Codegen](../new-architecture-library-intro#configure-codegen).
+
+The following options are available:
+
+- `--path` - Path to `package.json`. The default path is current working directory.
+- `--platform` - Target platform. Supported values: `android`, `ios`, `all`. The default value is `all`.
+- `--outputPath` - Path to override `codegenCofig.outputDir`. The default value is `undefined`.
+
+#### EXAMPLES
+
+- Read `package.json` from the current working directory, generate code based on its `codegenConfig`.
+
+```sh
+npx react-native codegen
+```
+
+- Read `package.json` from the current working directory, generate iOS code in the location defined in the `codegenConfig`.
+
+```sh
+npx react-native codegen --platform ios
+```
+
+- Read `package.json` from `third-party/some-library`, generate Android code in `third-party/some-library/android/generated`.
+
+```sh
+npx react-native codegen \
+    --path third-party/some-library \
+    --platform android \
+    --outputPath third-party/some-library/android/generated
+```
+
+## Including Generated Code into Libraries
+
+If you are working on a library, you can include codegen artifacts in it. Here is how you can do it:
+
+- Define `includesGeneratedCode: true` and `outputDir` in the `codegenConfig`, as described in [Configure Codegen](../new-architecture-library-intro#configure-codegen).
+- Run codegen locally with [the codegen CLI](pillars-codegen#the-codegen-cli).
+- Update your `package.json` to include generated code.
+- Update your [podspec](pillars-turbomodules#ios-create-the-podspec-file) to include generated code.
+- Update your [build.gradle file](../the-new-architecture/pillars-turbomodules#the-buildgradle-file) to include generated code.
+
+This setup has a number of benefits:
+
+- No need to rely on the app to run the codegen for you, the generated code is always there.
+- The implementation files are always consistent with the generated interfaces.
+- No need to [include two sets of files](backward-compatibility-turbomodules#android-1) to support the old and the new architecture on Android. You can only keep the new architecture one, and it is guaranteed to be backwards compatible.
+- No need to worry about codegen version mismatch between what is used by the app, and what was used during library development.
+- Since all native code is there, it is possible to ship the native part of the library as a prebuild.
```

Comment:
I would move the benefits as first paragraph below the `## Including Generated Code into Libraries`.